models/option.go

```diff
@@ -61,6 +93,12 @@ func (option *OptionNode) IsEmpty() bool {
 	return !option.IsValueOption() && !option.IsConfigOption()
 }
 
+// SetType ...
+func (option *OptionNode) SetType(optionType Type) *OptionNode {
```

Comment:
Note:
If the function already returns the OptionNode at the end, then you could use a value receiver instead.

Reply:
the reason I left it like this because then use it in a chained way and a single set way, 
```
myoption.SetType(...).DoOtherThing()
but also
myoption.SetType(...)
myoption.DoOtherThing()
```